### ISSUE TITLE
[Feature] Signal Project Tracking

### ISSUE THREAD

**Issue by amenity** [cityofaustin/atd-data-tech#3974]
Request by Renée O., AMD:

> Right now we don’t have a way to track signal modifications.  At this time we have some substantial projects going on at existing signals.  The signal status at these locations is shown as “turned on” and there is no acknowledgement of the modification project.  Many of these projects require contractor, inspection and signal tech participation.

## Requirements 
- Add signals that are tracked in Data Tracker as project components in Moped
- Sending users from within a Moped project to a list of signals(related to this project) in Data Tracker
- Storing a Moped Projects table for AMD-entered projects into an object in AMD Data Tracker, then relating it with AMD Data Tracker Work Orders
- Nice to have: [Data Tracker signal status](https://user-images.githubusercontent.com/1463708/109659817-b841d900-7b2d-11eb-81ed-6482121180f8.png) is easily visible in Moped, e.g. row icon or map pin changes color

**Out-of-scope:**
- Maintaining signal information in Moped
- Work-order-y integration #3669 

## Resources
- [Projected Signal Construction - Modifications - Short Term Adjustments](https://docs.google.com/spreadsheets/d/1pz1MYA1jLHUQtwmIfjZnazMmuCeFmpI07QV3Rjbh9mA/edit#gid=2046262291)
- [Renée's mockup of a Data Tracker project tracking feature](https://docs.google.com/spreadsheets/d/1mb6mCReEZHOE_wstWcLn8UufyLaebxQIwS9VV2or6Qk/edit#gid=88182665)
- [List Renée sent to ATSD for incorporation into the IMPDB](https://github.com/cityofaustin/atd-data-tech/files/6187368/Signal.Projects.2021.1.xlsx)
- [Example Estimate (ATS Contract)](https://docs.google.com/spreadsheets/d/1cbA5-7VkAOvJvsuiFOOpBngErHLyJy6c/edit#gid=324556068)
- [AMD Cost Estimate - Signal Work - Excel template](https://docs.google.com/spreadsheets/d/1fc9l5r39tqeUV6klEtjOSsarYnpsMccK/edit#gid=690637486) 
- ATD AMD Design and Construction Services Group meeting agendas: [4/29/21](https://drive.google.com/file/d/1WSSHNZNLRNSFRDFSau4tlCqbrpHtn8P3/view?usp=sharing) | [9/17/20](https://drive.google.com/file/d/1GGXNADn3yB-upz6Da3MP-_xdI4GIkQfx/view?usp=sharing)

**Additional notes:**
Goal is to avoid hitting the Knack API on Data Tracker's side too much; instead, we would only call the Knack Data Tracker API when a new project is added within Moped by an AMD user.

**Comment by amenity** [comment]
Getting more info from Renée via email.

**Comment by johnclary** [comment]
Follow-up from Renee:

> How are you tracking the information on contractor, inspection, and signal tech participation currently? If you have a doc to share that would be great. 

We track some this information in a weekly Design & Construction (D&C) meeting with the attached agenda. To my knowledge there is no place where inspectors are specifically identified with a project.  I have highlighted the signal modifications in yellow in this attachment so you can get an idea of the scale of magnitude.  There is a weekly meeting to schedule the tech’s work which generates the attached Traffic Signal Schedule document.  I am not sure how else they may be scheduled.

> Are these one-off projects, or are multiple signals done at once by the same contractor?
> Roughly how many of these are we talking about per month/year?  

It is hard to say how many signal mods we have annually.  Right now we have quite a few that are part of bond corridor or Active Transportation projects.  Many times our crews and our contractors are working alongside bond of Active Trans contactors/inspectors. 

We do have signal mods that are driven by development and other changes.  These could be considered more of a one-off in that they typically affect one intersection at a time (as opposed to a bond or Active Trans project which could affect a number of signals along a route).  However grouped together they are not uncommon or isolated occurrences.   

Brian/Jen/Scott – please feel free to jump in with any other thoughts, comments. etc.

**Comment by johnclary** [comment]
This overlaps with #881 (Extend work orders to other AMD workgroups) and the project tracking we plan to capture in Moped. For now, I think we just need to track and revisit the requirements in this issue as we close in on the Moped alpha release.

**Comment by amenity** [comment]
@zappfyllc assigning this to you because we discussed that earlier, but John and I also gave Nathan our blessing to pitch the Interim Project Database to AMD a second time. Can discuss more next week.

**Comment by zappfyllc** [comment]
@amenity I'm just waiting to hear how it goes with Nathan and AMD, and then I can put forth a diagram for the AMD Signals-Moped integration we've imagined based on how that goes. Otherwise, if Nathan onboards AMD to the Interim Project Database, I suggest we first evaluate his approach because we could just duplicate instead of coming up with our own implementation.

**Comment by amenity** [comment]
@zappfyllc - @johnclary pointed out that we have a [feature request](https://github.com/cityofaustin/atd-data-tech/issues/1921) that has overlap with this Moped functionality.

**Comment by johnclary** [comment]
@amenity do we have a Moped Epic that we could stick this in? My search didn't turn up a "Data Tracker Integration" epic, which would probably be the best place.

**Comment by amenity** [comment]
@johnclary - this was meant to be the parent Moped epic. But we can rearrange as you see fit!

**Comment by johnclary** [comment]
That works, thanks!

**Comment by amenity** [comment]
@janetchuhl @patrickm02L — if @janetchuhl has time while she's with us, documenting the existing signal tracking workflow in Data Tracker would be really helpful to tee this feature up for Moped 1.0.

**Comment by johnclary** [comment]
FWIW @dianamartin and I have been working with Renee to propose some technical changes to the way the signal lifecycle workflow is managed in the Data Tracker. [These slides](https://docs.google.com/presentation/d/1sqqugoSPbSnINWnrhULs4szBu9VUHbwXPGKMqTWK-HI/edit#slide=id.gdba6a68970_0_0) might be helpful.

**Comment by amenity** [comment]
Outdated-but-apparently-not-obsolete mockup: 
![Screen Shot 2021-07-23 at 1 02 03 PM](https://user-images.githubusercontent.com/1463708/126822864-623bad05-533f-4056-8f85-cb9d420cb632.png)
([Balsamiq](https://balsamiq.cloud/senhupm/p9y6tuy/rA62F))

**Comment by amenity** [comment]
Notes from today's meeting with @zappfyllc @johnclary Renée, and Ivonne to review first iteration of signal dashboard. 

**What projects should show on dashboard?**

-   Anything with a signal (like it does currently); doesn't need to be dependent on sponsor/funding/etc. 

-   E.g. Project sponsor (entity owning/justifying project) could be SRTS but AMD is executing

**Misc**

-   Good that we can handle multiple signals in a single project  

**Status updates/notes**

-   Ideal future state, inspector/designer is doing project updates ahead of meeting

-   Renée should be able to look at the project in Moped and report on it

-   On dashboard, show status update (like it does currently) - rename column to "Status update"

-   Not necessarily a must-have that notes can be edited/added in place on the table

**Funding**

-   Current FDU field won't be very useful for Ivonne since there is so much room for error; really likes the "Unit" FDU field John implemented in Data Tracker
<img width="653" alt="Screen Shot 2021-11-16 at 11 49 02 AM" src="https://user-images.githubusercontent.com/1463708/142041885-9a370658-ec2e-44d3-b59d-3f0a4c8517ad.png">



-   See [[Feature] Financial Code Tracking](https://github.com/cityofaustin/atd-data-tech/issues/4530) for backlog feature

**Files**

-   Difference between files related to signal asset and files related to the project; don't want to clutter Moped or have redundant data

-   Technicians need to know where to get project drawings; should be stored in Moped but/and easily accessible to technicians

-   Idea: Once a project is done, files are uploaded to Data Tracker signal with a column for which project they came from

**Moped 1.0 users**

-   Brian, Scott, inspectors, Billy's group

-   Technicians in the field will want to pull up project drawings
